pypose/optim/optimizer.py

```diff
@@ -451,17 +434,23 @@ def step(self, input, target=None, weight=None):
             <https://github.com/pypose/pypose/tree/main/examples/module/pgo>`_.    
         '''
         for pg in self.param_groups:
+            weight = self.weight if weight is None else weight
             R = self.model(input, target, weight)
             J = modjac(self.model, input=(input, target, weight), **self.jackwargs)
             R, J = self.corrector(R = R, J = J)
             self.last = self.loss = self.loss if hasattr(self, 'loss') \
                                     else self.model.loss(input, target, weight)
-            A, self.reject_count = J.T @ J, 0
+            J_T = J.reshape(tuple(R.shape)+(-1,))
+            J_T = J_T.permute([len(J_T.shape)-1,]+list(range(len(J_T.shape)-1)))
```

Comment:
why do you want the dimension of J to be rotated?
and `len(J_T.shape)` is `J_T.ndim`

Reply:
> len(

![Screenshot from 2022-12-17 16-59-34](https://user-images.githubusercontent.com/17938325/208234399-44161406-b4dd-4c7d-b939-7f22e1a13e75.png)